dashboard/test/models/section_hidden_stage_test.rb

```diff
@@ -0,0 +1,18 @@
+require 'test_helper'
+
+class SectionHiddenStageTest < ActiveSupport::TestCase
+  test "can only manage SectionHiddenStage for section you own" do
+    teacher1 = create(:teacher)
+    teacher2 = create(:teacher)
+    student = create(:student)
+
+    stage = create :stage
+
+    # Create section owned by teacher 1.
+    section = Section.create!(user: teacher1, name: "section 1")
+
+    assert Ability.new(teacher1).can?(:manage, SectionHiddenStage.create(stage_id: stage.id, section_id: section.id))
+    assert !Ability.new(teacher2).can?(:manage, SectionHiddenStage.create(stage_id: stage.id, section_id: section.id))
+    assert !Ability.new(student).can?(:manage, SectionHiddenStage.create(stage_id: stage.id, section_id: section.id))
+  end
+end
```

Comment:
Honestly, I agree (starting from a blank slate) that this makes more sense where you put at. But the slate isn't blank, so I think it makes sense to conform to existing test structure, putting it in `AbilityTest`. Feel free to keep or move.